src/main/java/controllers/visualizations/BarChartController.java

```diff
@@ -0,0 +1,175 @@
+
+
+	/**
+	 * Create new BarChartController object.
+	 *
+	 * @param table the table for the graph.
+	 * @param vBox  the box for the input comboBoxes.
+	 */
+	public BarChartController(DataTable table, VBox vBox) {
+		this.table = table;
+		this.vBox = vBox;
+	}
+
+	/**
+	 * Initialize the fxml objects for the barChart.
+	 */
+	public void initializeFields() {
+		xAxisBox = new ComboBox<>();
+		xAxisErrorLabel = new Label();
+		xAxisErrorLabel.setMaxWidth(Double.MAX_VALUE);
+		yAxisBox = new ComboBox<>();
+		yAxisErrorLabel = new Label();
+		yAxisErrorLabel.setMaxWidth(Double.MAX_VALUE);
+		setColumnDropDown(xAxisBox, table);
+		setColumnDropDown(yAxisBox, table);
+	}
+
+	/**
+	 * Sets the Listener for the xAxis ComboBox.
+	 * When the value changes, the data of the selected column will be added to the
+	 * CategoryAxis of the barChart. An error message is shown to the user when the column
+	 * does not contain appropriate values for a CategoryAxis.
+	 */
+	public void setXAxisEventListener() {
+		xAxisBox.valueProperty().addListener((observable1, oldValue1, newValue1) -> {
+			xCol = newValue1;
+			List<String> dataxcol = table.getRows().stream()
+					.map(row -> row.getValue(xCol).toString())
+					.collect(Collectors.toList());
+			try {
+				xAxis = new CategoryAxis(FXCollections.observableArrayList(dataxcol));
+				setErrorLabel(xAxisErrorLabel, "");
+				xSet = true;
+			} catch (Exception e) {
+				setErrorLabel(xAxisErrorLabel, "Please select a column with distinct values.");
+			}
+		});
+	}
+
+	/**
+	 * Sets the Listener for the yAxis ComboBox.
+	 * The NumberAxis is created and the scale is set.
+	 */
+	public void setYAxisEventListener() {
+		dataColumnChangeListener = (observable, oldValue, newValue) -> {
+			yCol = newValue;
+			DataDescriber<NumberValue> yColDescriber = new RowValueDescriber<>(yCol);
+			try {
+				float max = new Maximum(table, yColDescriber).calculate().getValue();
+				float min = new Minimum(table, yColDescriber).calculate().getValue() - 1;
+				int sep = computeSeparatorValue(max, min);
+				yAxis = new NumberAxis(yCol.getName(), min, max, sep);
+				setErrorLabel(yAxisErrorLabel, "");
+				ySet = true;
+			} catch (Exception e) {
+				setErrorLabel(yAxisErrorLabel, "Please select a column with number values.");
+			}
+		};
+		yAxisBox.valueProperty().addListener(dataColumnChangeListener);
+	}
+
+	/**
+	 * Called to initialize the barChartController object.
+	 */
+	public void initialize() {
+		initializeFields();
+		setXAxisEventListener();
+		setYAxisEventListener();
+		vBox.getChildren().addAll(xAxisErrorLabel, xAxisBox, yAxisErrorLabel, yAxisBox);
+	}
+
+	/**
+	 * Compute approximately a tenth of the range of the axis.
+	 *
+	 * @param max the maximum value of the axis.
+	 * @param min the minimum value of the axis.
+	 * @return the computed seperator value as int.
+	 */
+	public int computeSeparatorValue(float max, float min) {
+		return (int) (max - min) / YAXIS_SEPARATION;
```

Comment:
Hier is Math.round eigenlijk beter, dan verlies je geen precisie of anders Math.floor.